### PR TITLE
Issue #13682: Create checks to place violations on missing macros in …

### LINKED ISSUE #13682: Create checks to place violations on missing macros in templates
From https://github.com/checkstyle/checkstyle/pull/13643#discussion_r1313201703
> Can we create a regexp module to place violations on <source> tags in templates, or something similar to catch these?

We need a check to ensure that no examples are shown in templates without being tested. For that, we can place violations on source tags in templates.

### PR DESCRIPTION
**Pull request by nrmancuso**
Closes #13682

### CODE REVIEW

File: config/checkstyle-non-main-files-suppressions.xml [hunk 1]
```diff
@@ -167,4 +167,94 @@
   <!-- We use negative suppression to check workflow files only. -->
   <suppress id="workflowJobStepSpacing" files="^.((?!github[\\/]workflows[\\/]).)*$" />
 
+  <!-- Examples macro suppressions -->
```
Comment (nrmancuso) [inline config/checkstyle-non-main-files-suppressions.xml:170]: We can create one issue per section here to get macros in place for each section here if we agree to this.

File: config/checkstyle-non-main-files-suppressions.xml [hunk 2]
```diff
@@ -167,4 +167,94 @@
...
+  <suppress id="exampleMacroMustExist"
+         files="src[\\/]xdocs[\\/]filters[\\/]suppressionxpathfilter.xml.template"/>
+
+  <!-- Parent module macro suppressions -->
+  <suppress id="parentModuleMacroMustExist"
+         files="src[\\/]xdocs[\\/]checks[\\/]coding[\\/]unnecessarysemicolonafteroutertypedeclaration.xml.template"/>
+  <suppress id="parentModuleMacroMustExist"
+         files="src[\\/]xdocs[\\/]checks[\\/]coding[\\/]unnecessarysemicolonaftertypememberdeclaration.xml.template"/>
+
+  <!-- Violations messages macro suppressions -->
```
Comment (nrmancuso) [inline config/checkstyle-non-main-files-suppressions.xml:214]: Obviously some of these are still in the works (suppressions and filters), but it looks like we missed some checks here.

**Review by nrmancuso — COMMENTED** [review]
Comments:

**Comment by nrmancuso** [comment]
I am going to add another module for properties right now, I missed this one.

**Review by romani — APPROVED** [review]
throught a time ....

I found this model is good only is migration will take months and years and contributors arenot aware and need guideance from build violations.

If migration planned to be quick, it only create inconvenience and conflicts in suppression files. 
in last PR from @stoyanK7 , I even suggested to not do this model of suppression for not-ye-migrated and simply grep content of file to decided to do validation or not.

**Comment by nrmancuso** [comment]
I mostly agree with sentiments at https://github.com/checkstyle/checkstyle/pull/13725#pullrequestreview-1618827729, except that we will probably end up with some suppressions that are around for a long time, like filters and ...

**Review by rnveach — APPROVED** [review]

**Comment by rnveach** [comment]
Parts of CI are failing

**Comment by nrmancuso** [comment]
@rnveach failing CI is not related, looks like Mac is builds on azure have not been stable for the last day or two.

**Comment by romani** [comment]
I double checked, looks unrelated CI service issues.

**Pull request merged**